ui/components/app/assets/asset-list/asset-list-control-bar/asset-list-control-bar.tsx

```diff
@@ -65,19 +74,36 @@ const AssetListControlBar = ({
   const t = useI18nContext();
   const dispatch = useDispatch();
   const trackEvent = useContext(MetaMetricsContext);
+  const history = useHistory();
   const popoverRef = useRef<HTMLDivElement>(null);
   const currentNetwork = useSelector(getCurrentNetwork);
+  const useNftDetection = useSelector(getUseNftDetection);
   const allNetworks = useSelector(getNetworkConfigurationsByChainId);
   const isTokenNetworkFilterEqualCurrentNetwork = useSelector(
     getIsTokenNetworkFilterEqualCurrentNetwork,
   );
+  const isMainnet = useSelector(getIsMainnet);
+  const allChainIds = useSelector(getAllChainsToPoll);
+
+  const { collections } = useNftsCollections();
 
   const tokenNetworkFilter = useSelector(getTokenNetworkFilter);
   const [isTokenSortPopoverOpen, setIsTokenSortPopoverOpen] = useState(false);
   const [isImportTokensPopoverOpen, setIsImportTokensPopoverOpen] =
     useState(false);
   const [isNetworkFilterPopoverOpen, setIsNetworkFilterPopoverOpen] =
     useState(false);
+  const [isImportNftPopoverOpen, setIsImportNftPopoverOpen] = useState(false);
+
+  const shouldShowRefreshButtons = useMemo(
+    () => isMainnet || Object.keys(collections).length > 0,
+    [isMainnet, collections],
+  );
+
+  const shouldShowEnableAutoDetect = useMemo(
+    () => isMainnet && !useNftDetection,
```

Comment:
Can we add linea mainnet to this check too; i think we had `isLineaMainnet`

Reply:
https://github.com/MetaMask/metamask-extension/pull/32020/commits/adc7e1b473f2b9d1674e82cd074f8b76a678ad4e